plasma_framework/test/src/exits/payment/PaymentInFlightExitable.test.js

```diff
@@ -0,0 +1,458 @@
+                const inputTx2 = createInputTransaction(DUMMY_INPUT_2, bob, AMOUNT);
+                const inputTx3 = createInputTransaction(DUMMY_INPUT_3, alice, AMOUNT);
+
+                const inputUtxosPos = [buildUtxoPos(BLOCK_NUMBER, 0, 0), buildUtxoPos(BLOCK_NUMBER, 1, 0)];
+                const inFlightTx = createInFlightTx([inputTx1, inputTx2], inputUtxosPos, carol, AMOUNT);
+
+                const { args, inputTxsBlockRoot } = buildIfeStartArgs([inputTx2, inputTx3], inputUtxosPos, inFlightTx);
+
+                await this.framework.setBlock(BLOCK_NUMBER, inputTxsBlockRoot, 0);
+
+                await expectRevert(
+                    this.exitGame.startInFlightExit(args, { from: alice, value: IN_FLIGHT_EXIT_BOND }),
+                    'In-flight transaction does not stem from input transactions',
+                );
+            });
+
+            it('should fail when number of input transactions does not match number of input utxos positions', async () => {
+                const inputTx1 = createInputTransaction(DUMMY_INPUT_1, alice, AMOUNT);
+                const inputTx2 = createInputTransaction(DUMMY_INPUT_2, bob, AMOUNT);
+
+                const inputUtxosPos = [buildUtxoPos(BLOCK_NUMBER, 0, 0)];
+                const inFlightTx = createInFlightTx([inputTx1, inputTx2], inputUtxosPos, carol, AMOUNT);
+
+                const { args, inputTxsBlockRoot } = buildIfeStartArgs([inputTx1, inputTx2], inputUtxosPos, inFlightTx);
+
+                await this.framework.setBlock(BLOCK_NUMBER, inputTxsBlockRoot, 0);
+
+                await expectRevert(
+                    this.exitGame.startInFlightExit(args, { from: alice, value: IN_FLIGHT_EXIT_BOND }),
+                    'Number of input transactions does not match number of provided input utxos positions.',
+                );
+            });
+
+            it('should fail when number of input transactions does not match in-flight transactions number of inputs', async () => {
+                const inputTx1 = createInputTransaction(DUMMY_INPUT_1, alice, AMOUNT);
+                const inputTx2 = createInputTransaction(DUMMY_INPUT_2, bob, AMOUNT);
+
+                const inputUtxosPos = [buildUtxoPos(BLOCK_NUMBER, 0, 0), buildUtxoPos(BLOCK_NUMBER, 1, 0)];
+                const inFlightTx = createInFlightTx([inputTx1], inputUtxosPos, carol, AMOUNT);
+
+                const { args, inputTxsBlockRoot } = buildIfeStartArgs([inputTx1, inputTx2], inputUtxosPos, inFlightTx);
+
+                await this.framework.setBlock(BLOCK_NUMBER, inputTxsBlockRoot, 0);
+
+                await expectRevert(
+                    this.exitGame.startInFlightExit(args, { from: alice, value: IN_FLIGHT_EXIT_BOND }),
+                    'Number of input transactions does not match number of in-flight transaction inputs.',
+                );
+            });
+
+            it('should fail when in-flight transaction overspends - single token input/output', async () => {
```

Comment:
minor: I would say `.... given single token input/output in tx`

Reply:
done